client/ayon_core/pipeline/actions.py

```diff
@@ -10,6 +13,224 @@
 from .load.utils import get_representation_path_from_context
 
 
+class LauncherActionSelection:
+    """Object helper to pass selection to actions.
+
+    Object support backwards compatibility for 'session' from OpenPype where
+    environment variable keys were used to define selection.
+
+    Args:
+        project_name (str): Selected project name.
+        folder_id (str): Selected folder id.
+        task_id (str): Selected task id.
+        folder_path (Optional[str]): Selected folder path.
+        task_name (Optional[str]): Selected task name.
+        project_entity (Optional[dict[str, Any]]): Project entity.
+        folder_entity (Optional[dict[str, Any]]): Folder entity.
+        task_entity (Optional[dict[str, Any]]): Task entity.
+
+    """
+    def __init__(
+        self,
+        project_name,
+        folder_id,
+        task_id,
+        folder_path=None,
+        task_name=None,
+        project_entity=None,
+        folder_entity=None,
+        task_entity=None
```

Comment:
Could you point me to where we're actually setting these from the launcher? I can't seem to find that.

Reply:
Nowhere.

Reply:
In that case - we shouldn't expose it as if it'll be available? :) Or will you be adding commits to fix it? :)

Reply:
Ah wait, I only _now_ see that it queries it whenever you access it. Nevermind. I'm blind.

Reply:
Why not? It might be used for testing.